plugins/vpp/l3plugin/arp_config_test.go

```diff
@@ -0,0 +1,192 @@
+	RegisterTestingT(t)
+	ctx := &vppcallmock.TestCtx{
+		MockVpp: &mock.VppAdapter{},
+	}
+	connection, _ := core.Connect(ctx.MockVpp)
+	defer connection.Disconnect()
+
+	plugin := &l3plugin.ArpConfigurator{}
+
+	// Test init
+	err := plugin.Init(logging.ForPlugin("test-log", logrus.NewLogRegistry()), connection, nil, false)
+	Expect(err).To(BeNil())
+	err = plugin.Close()
+	Expect(err).To(BeNil())
+}
+
+var arpTableEntries = []l3.ArpTable_ArpEntry{
+	{
+		Interface:   "tap1",
+		IpAddress:   "192.168.10.21",
+		PhysAddress: "59:6C:DE:AD:00:01",
+		Static:      true,
+	},
+	{
+		Interface:   "tap2",
+		IpAddress:   "192.168.10.22",
+		PhysAddress: "59:6C:DE:AD:00:02",
+		Static:      true,
+	},
+	{
+		Interface:   "tap3",
+		IpAddress:   "dead::01",
+		PhysAddress: "59:6C:DE:AD:00:03",
+		Static:      false,
+	},
+	{
+		Interface:   "tap4",
+		IpAddress:   "dead::02",
+		PhysAddress: "59:6C:DE:AD:00:04",
+		Static:      false,
+	},
+}
+
+// Test adding of ARP entry
+func TestAddArp(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := arpTestSetup(t)
+	defer arpTestTeardown(connection, plugin)
+
+	err := plugin.AddArp(&arpTableEntries[2])
+	Expect(err).To(BeNil())
+
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err = plugin.AddArp(&arpTableEntries[0])
+	Expect(err).To(BeNil())
+
+	// Test isValidARP
+	err = plugin.AddArp(&l3.ArpTable_ArpEntry{})
+	Expect(err).To(Not(BeNil()))
+	err = plugin.AddArp(&l3.ArpTable_ArpEntry{Interface: "tap5"})
+	Expect(err).To(Not(BeNil()))
+	err = plugin.AddArp(&l3.ArpTable_ArpEntry{Interface: "tap6", IpAddress: "192.168.10.33"})
+	Expect(err).To(Not(BeNil()))
+}
+
+// Test deleting of ARP entry
+func TestDeleteArp(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := arpTestSetup(t)
+	defer arpTestTeardown(connection, plugin)
+
+	// delete arp for non existing intf
+	err := plugin.DeleteArp(&arpTableEntries[3])
+	Expect(err).To(BeNil())
+
+	// delete arp-chached ARP entry
+	err = plugin.AddArp(&arpTableEntries[2])
+	Expect(err).To(BeNil())
+	err = plugin.DeleteArp(&arpTableEntries[2])
+	Expect(err).To(BeNil())
+
+	// delete added ARP entry
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err = plugin.AddArp(&arpTableEntries[0])
+	Expect(err).To(BeNil())
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err = plugin.DeleteArp(&arpTableEntries[0])
+	Expect(err).To(BeNil())
+}
+
+// Test changing of ARP entry
+func TestChangeArp(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := arpTestSetup(t)
+	defer arpTestTeardown(connection, plugin)
+
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err := plugin.AddArp(&arpTableEntries[0])
+	Expect(err).To(BeNil())
+
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err = plugin.ChangeArp(&arpTableEntries[1], &arpTableEntries[0])
+	Expect(err).To(BeNil())
+}
+
+// Test resolving of created ARPs
+func TestArpResolveCreatedInterface(t *testing.T) {
```

Comment:
this test is not correct. ResolveCreatedInterface is called by interface registration event, but even if you call the method directly, you have to register the interface to mapping.

ResolveCreatedInterface should find the created interface, configure arp and remove it from the cache, which does not happen here. If you would test internal state, you would fount it yourself.